src/sparse/KokkosSparse_sptrsv.hpp

```diff
@@ -210,10 +219,584 @@ namespace Experimental {
     BType_Internal b_i = b;
     XType_Internal x_i = x;
 
-    KokkosSparse::Impl::SPTRSV_SOLVE<const_handle_type, RowMap_Internal, Entries_Internal, Values_Internal, BType_Internal, XType_Internal>::sptrsv_solve (&tmp_handle, rowmap_i, entries_i, values_i, b_i, x_i);
+    KokkosSparse::Impl::SPTRSV_SOLVE<const_handle_type, RowMap_Internal, Entries_Internal, Values_Internal, BType_Internal, XType_Internal>::
+      sptrsv_solve (&tmp_handle, rowmap_i, entries_i, values_i, b_i, x_i);
 
   } // sptrsv_solve
 
+
+#ifdef KOKKOSKERNELS_ENABLE_TPL_SUPERLU
+  // ---------------------------------------------------------------------
+  template <typename scalar_type,
+            typename ordinal_type,
+            typename size_type,
+            typename KernelHandle,
+            typename execution_space      = Kokkos::DefaultExecutionSpace,
+            typename host_execution_space = Kokkos::DefaultHostExecutionSpace>
+  void sptrsv_symbolic(
+      KernelHandle *kernelHandleL,
+      KernelHandle *kernelHandleU,
+      SuperMatrix &L,
+      SuperMatrix &U)
+  {
+    typedef KokkosSparse::CrsMatrix<scalar_type, ordinal_type, host_execution_space, void, size_type> host_crsmat_t;
+    typedef KokkosSparse::CrsMatrix<scalar_type, ordinal_type,      execution_space, void, size_type> crsmat_t;
+
+    typedef typename host_crsmat_t::StaticCrsGraphType host_graph_t;
+    typedef typename      crsmat_t::StaticCrsGraphType      graph_t;
+
+    Kokkos::Timer timer;
+    Kokkos::Timer tic;
+
+    // ===================================================================
+    // load sptrsv-handles
+    auto *handleL = kernelHandleL->get_sptrsv_handle ();
+    auto *handleU = kernelHandleU->get_sptrsv_handle ();
+    int *etree = handleL->get_etree ();
+
+    // ===================================================================
+    // load options
+    bool col_majorL = handleL->is_column_major ();
+    bool col_majorU = handleU->is_column_major ();
+    bool merge = handleL->get_merge_supernodes ();
+    bool UinCSC = handleU->is_column_major ();
+    bool needEtree = (handleL->get_algorithm () == SPTRSVAlgorithm::SUPERNODAL_SPMV ||
+                      handleL->get_algorithm () == SPTRSVAlgorithm::SUPERNODAL_ETREE);
+    if (needEtree && etree == nullptr) {
+      std::cout << std::endl
+                << " ** etree needs to be set before calling sptrsv_symbolic with SuperLU **"
+                << std::endl << std::endl;
+      return;
+    }
+
+    // ===================================================================
+    // read CrsGraph from SuperLU factor
+    #ifdef KOKKOS_SPTRSV_SUPERNODE_PROFILE
+    std::cout << " > Read SuperLU factor into KokkosSparse::CrsMatrix (invert diagonal and copy to device)" << std::endl;
```

Comment:
1. The name "PROFILE" implies performance profiling, not printing verbose debugging output.
2. Why is this a compile-time option?  Why not make it a run-time option?  You could store in the handle a `bool` called "`verbose`" that tells functions to print verbose output like this.